mmseg/apis/test.py

```diff
@@ -61,13 +64,35 @@ def single_gpu_test(model, data_loader, show=False, out_dir=None):
                     show=show,
                     out_file=out_file)
 
+        if isinstance(result, list):
+            if efficient_test:
+                tmp_result = []
+                for item in result:
+                    temp_file_name = tempfile.NamedTemporaryFile(
+                        suffix='.npy', delete=False).name
+                    np.save(temp_file_name, item)
```

Comment:
I suggest making this part a function. So that we could have sth like:
```
results.extend(np2tmp(_) for _ in result)
```

Reply:
Is there a pair of []?
```python
results.extend([np2tmp(_) for _ in result])
```